### PR TITLE
Export Transactions as CSV

### LINKED ISSUE #912: Add transactional data export
Add a menu-bar item that allows users to export their transactional data over a given period of time and put it nicely into a csv for use elsewhere.

As a first pass this should be able to export information about tickets and votes that can be used for tax filing purposes.

Please update this issue with information that you'd like to see.

### PR DESCRIPTION
**Pull request by matheusd**
Fix #912 

POC for the moment. No fancy ui yet, only the export function on the "transactions" tab. Will create a local "test.csv" file.

### CODE REVIEW

**Comment by alexlyp** [comment]
Great stuff!  Looks really good. 
I'm trying to figure out what we should do with addresses.  I go back and forth whether those are relevant for this type of data dump, or is merely having the overall transaction in/out is enough.

**Comment by matheusd** [comment]
I think for a csv export, the addresses aren't important (as there is likely to be multiple, you'd have to also signal which are from the wallet, external vs internal, etc). We could write a json export if someone wants to process more stuff (but then, they could just use dcrctl anyway).

**Comment by matheusd** [comment]
I'll wait for the resolution of #1135 so I can include its file picker to select the destination file.

**Comment by matheusd** [comment]
Ready for final review. Added the following:

- Daily balances stats and export (hopefully to be used as source for  the overview graphs)
- FIle input browser for the destination csv file

I had to slightly tweak the PathInputBrowser component to allow using a save dialog as well as the open dialog that @cipherzzz implemented. Also added the ability to use filters to it (so that the extension is appended by default when creating a file).

**Comment by matheusd** [comment]
Had to change the transaction gathering logic due to having to decode the full vote/revoke transactions to account for the correct commitment amounts in various situations.

**Comment by alexlyp** [comment]
Nice, I beileve the csv's properly match up now.  I'll do a review then should be good to go.

File: app/actions/StatisticsActions.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+    dispatch({filename: csvFilename, type: EXPORT_COMPLETED});
+    if (fd) {
+      fs.closeSync(fd);
+      fd = null;
+    }
+  };
+
+  dispatch(calcFunction({opts, startFunction, progressFunction, endFunction, errorFunction}));
+
+  // if (fd !== null) {
```
Comment (alexlyp) [inline app/actions/StatisticsActions.js]: looks like this can be removed

File: app/actions/StatisticsActions.js [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+      {name: "direction"},
+      {name: "fee", type: VALUE_TYPE_ATOMAMOUNT},
+      {name: "amount", type: VALUE_TYPE_ATOMAMOUNT},
+      {name: "credits", type: VALUE_TYPE_ATOMAMOUNT},
+      {name: "debits", type: VALUE_TYPE_ATOMAMOUNT},
+    ],
+  });
+
+  const formatTx = (tx) => {
+    // console.log(tx);
```
Comment (alexlyp) [inline app/actions/StatisticsActions.js]: remove

File: app/actions/StatisticsActions.js [hunk 3]
```diff
@@ -0,0 +1,289 @@
...
+      direction: tx.direction,
+      fee: tx.fee,
+      amount: tx.amount,
+      credits: tx.tx.getCreditsList().reduce((s, c) => s + c.getAmount(), 0),
+      debits: tx.tx.getDebitsList().reduce((s, d) => s + d.getPreviousAmount(), 0),
+    };
+  };
+
+  const txDataCb = (mined) => {
+    //console.log(mined);
```
Comment (alexlyp) [inline app/actions/StatisticsActions.js]: another

**Pull request merged**